rfcs/2021-06-15-shipping-blade.md

```diff
@@ -0,0 +1,438 @@
+```
+
+#### Pros
+* Don't have to write the whole import statement for every component we want. If there's an import statement already we just add it to the existing named import statement
+
+```js
+// consumer/src/App.tsx
+import { Button, Text, overrideTheme, paymentTheme } from '@razorpay/blade'
+```
+
+#### Cons
+* Everytime we add anything we need to ensure that we re-export it from the top level re-exports.
+
+```js
+// packages/blade/index.ts
+export * from 'src/components/Button'
+export * from 'src/components/Text'
+export * from 'src/tokens/theme'
+```
+
+### 3. `import` from each category(components, tokens,types, utils) root package
+```js
+/* mono-package with per category import */
+import { Button, Text } from '@razorpay/blade/components'
+import { overrideTheme, paymentTheme } from '@razorpay/blade/theme'
+import { getColorScheme, useColorScheme } from '@razorpay/blade/utils'
+```
+
+#### Pros
+* Better readability and structured imports. We know what we want and from where will it come.
+* No long running import statements with concerns mixed
+
+```js
+import { Button, Text, overrideTheme, paymentTheme, getColorScheme, useColorScheme } from '@razorpay/blade'
+```
+
+#### Cons
+* We need to maintain top level re-exports per category
+```js
+// packages/blade/src/components/index.ts
+export * from 'src/components/Button'
+export * from 'src/components/Text'
+
+// packages/blade/src/tokens/theme/index.ts
+export * from 'src/tokens/theme/paymentTheme'
+export * from 'src/tokens/theme/bankingTheme'
+export * from 'src/tokens/theme/overrideTheme'
+
+// packages/blade/src/utils/index.ts
+export * from 'src/tokens/utils/getColorScheme'
+export * from 'src/tokens/utils/useColorScheme'
+```
+
+### 4. `import` named components from each category(components, tokens,types, utils) root package
+```js
+/* mono-package with per category per component import */
+import Button from '@razorpay/blade/components/Button'
+import Text from '@razorpay/blade/components/Text'
+import paymentTheme from '@razorpay/blade/theme/paymentTheme'
+```
+
+#### Pros
+* No need to maintain top level re-exports per category. Just write your components and be done with it.
+
+```js
+// packages/blade/src/components/index.ts
+export * from 'src/components/Button'
+export * from 'src/components/Text'
+
+// packages/blade/src/tokens/theme/index.ts
+export * from 'src/tokens/theme/paymentTheme'
+export * from 'src/tokens/theme/bankingTheme'
+export * from 'src/tokens/theme/overrideTheme'
+
+// packages/blade/src/utils/index.ts
+export * from 'src/tokens/utils/getColorScheme'
+export * from 'src/tokens/utils/useColorScheme'
+```
+
+#### Cons
+* We have to write the whole named `import` statement for everything we want to import and with every category
+
+### Which approach we'll take?
+We'll go with approach [#3](#3-import-from-each-categorycomponents-tokenstypes-utils-root-package)
```

Comment:
Can you re-iterate in few sentences why we should take 3rd approach?
You're directly reaching a conclusion without any inference.

Also, I think the above comparison pros/cons are not conclusive enough to infer to a conclusion.
We need to highlight pros/cons considering how it would impact consumers: easy upgrades, bundle size, tree shaking, etc.
All the cons are an internal implementation detail. We need to set up re-exports separately yes, but it's mostly one-time effort to create the right structure & index files.

Reply:
>Can you re-iterate in few sentences why we should take 3rd approach?
You're directly reaching a conclusion without any inference.

The pros and cons in each section are elaborative enough. Why to write things again in this section?

The re-exports is a challenging bit on the maintainer side. It might seem easy when you're reading this RFC but it becomes pain when you have it in codebase.

>We need to highlight pros/cons considering how it would impact consumers: easy upgrades, bundle size, tree shaking, etc.
The conclusion is for this section 🙂 . The other things you're talking about is covered in the next section. 

The RFC is splitted into sections with each question that it answers. Please go through the digram first. I know it's complicated hence I had put enough efforts in the diagram to make it simpler to consume for the readers.